### ISSUE TITLE
Investigate accessibility statements

### ISSUE THREAD

**Issue by gabalafou** [Quansight-Labs/jupyter-a11y-mgmt#52]
This Github issue came out of the Nov 11 team meeting: https://github.com/Quansight-Labs/jupyter-a11y-mgmt/issues/49

- Tania likes the [Gatsby docs accessibility statement](https://www.gatsbyjs.com/accessibility-statement/). 
- Gabriel added [Adrian Roselli's accessibility statement](https://adrianroselli.com/accessibility-statement)

**Comment by trallard** [comment]
I have shared this gov.uk resources before
- https://www.gov.uk/government/publications/sample-accessibility-statement
- https://www.gov.uk/guidance/accessibility-monitoring-how-we-test

@trallard and @isabela-pf  to work on a lightweight version for Jupyter

**Comment by isabela-pf** [comment]
Discussions were also happening at [jupyter/accessibility #72](https://github.com/jupyter/accessibility/issues/72)!

**Comment by isabela-pf** [comment]
Okay, here are some thoughts on accessibility statements based on all the resources previously shared.

## What we get from the [W3C accessibility statement generator](https://www.w3.org/WAI/planning/statements/generator/#create)

I started to make an outline, but the W3C's statement generator is a great baseline, so I've filled to the best of my abilitythat out for review below. I think we will want to edit and possibly add to it, though.

<!DOCTYPE html>
<h2> ⚠️ Draft Accessibility Statement for <span class="basic-information website-name">JupyterLab</span></h2>
<p>
	This is an accessibility statement from <span class="basic-information organization-name">Jupyter accessibility contributors</span>.
</p>
<h3>Measures to support accessibility</h3>
<p>
	<span class="basic-information organization-name">Jupyter accessibility contributors</span>
	take the following measures to ensure accessibility of
	<span class="basic-information website-name">JupyterLab</span>:
</p>
<ul class="organizational-effort accessibility-measures">
	<li>Include accessibility as part of our mission statement.</li>
	<li>Provide continual accessibility training for our community.</li>
	<li>Assign clear accessibility goals and responsibilities.</li>
	<li>Employ formal accessibility quality assurance methods.</li>
</ul>
<h3>Conformance status</h3>
<p>
	The <a href="https://www.w3.org/WAI/standards-guidelines/wcag/">Web Content Accessibility Guidelines (WCAG)</a> defines requirements for designers and developers to improve accessibility for people with disabilities. It defines three levels of conformance: Level A, Level AA, and Level AAA.
	<span class="basic-information website-name">JupyterLab</span>
	is
	<span class="basic-information conformance-status" data-printfilter="lowercase">non conformant</span>
	with
	<span class="basic-information conformance-standard"><span data-negate="">WCAG 2.1 level AA</span>.</span>
	<span>
	<span class="basic-information conformance-status">Non conformant</span>
	means that
	<span class="basic-information conformance-meaning">the content does not conform the accessibility standard</span>.
</span>
</p>
<h3>Feedback</h3>
<p>
	We welcome your feedback on the accessibility of
	<span class="basic-information website-name">JupyterLab</span>.
	Please let us know if you encounter accessibility barriers on
	<span class="basic-information website-name">JupyterLab</span>:
</p>
<ul class="basic-information feedback h-card">
    <li class="contact-other p-note"><a href="https://github.com/jupyter/accessibility/issues/new">Write an issue on jupyter/accessibility</a><br><a href="https://github.com/jupyterlab/jupyterlab/issues/new">Write an issue on jupyterlab/jupyterlab</a> and request it be labeled <a href="https://github.com/jupyterlab/jupyterlab/labels/tag%3AAccessibility">tag:accessibility</a></li>
</ul>
<h3>Compatibility with browsers and assistive technology</h3>
<p>
	<span class="basic-information website-name">JupyterLab</span>
	is designed to be compatible with the following technologies:
</p>
<ul class="technical-information compatible-environments">
	<li> Windows, macOS, iOS, Android; Firefox, Chrome, Safari, Chromium browsers (mobile and desktop).</li>
</ul>
<p>
	<span class="basic-information website-name">JupyterLab</span>
	is not compatible with:
</p>
<ul class="technical-information incompatible-environments">
	<li>Internet Explorer, Edge; JAWS, NVDA, VoiceOver, Narrator, Orca screen readers; voice control technology.</li>
</ul>
<h3>Technical specifications</h3>
<p>
	Accessibility of
	<span class="basic-information website-name">JupyterLab</span>
	relies on the following technologies to work with the particular combination of web browser and any assistive technologies or plugins installed on your computer:
</p>
<ul class="technical-information technologies-used">
	<li>HTML</li>
	<li>WAI-ARIA</li>
	<li>CSS</li>
	<li>JavaScript</li>
</ul>
<p>These technologies are relied upon for conformance with the accessibility standards used.</p>
<h3>Limitations and alternatives</h3>
<p>
	Despite our best efforts to ensure accessibility of
	<span class="basic-information website-name">JupyterLab</span>, there may be some limitations. Below is a description of known limitations, and potential solutions. Please contact us if you observe an issue not listed below.
</p>
<p>
	Known limitations for
	<span class="basic-information website-name">JupyterLab</span>:
</p>
<ol class="technical-information accessibility-limitations">
    <li><strong>Documents</strong>: Documents writen by the community may not include accessible content because we do not and cannot review every document that can be opened in JupyterLab. To support accessible documents, we are drafting guidelines for accessible document content with an emphasis on Jupyter notebooks. Please report the issue to the author and <a href="https://github.com/jupyter/accessibility/issues/new">open an issue on jupyter/accessibility</a> describing the problem and the behavior you expect so we may integrate it into our content guidelines.</li>
    <li><strong>JupyterLab extensions</strong>: JupyterLab extensions writen by the community may not be accessible because JupyterLab extensions can be written by anyone in the community and have no required review process. We do not and can not review every JupyterLab extension. To support accessible extensions, we encourage extension authors to use existing, accessible JupyterLab components for their extension. We also  provide periodic opportunities for community education on accessibility. Please report the issue to the author and let them know the <a href="https://github.com/jupyter/accessibility/">jupyter/accessibility </a>community may be able to provide guidance.</li>
</ol>
<h3>Assessment approach</h3>
<p>
	<span class="basic-information organization-name">Jupyter accessibility contributors</span>
	assessed the accessibility of
	<span class="basic-information website-name">JupyterLab</span>
	by the following approaches:
</p>
<ul class="technical-information assessment-approaches">
	<li>Self-evaluation</li>
    <li>Regular automated testing to monitor for regressions (can be found at <a href="https://github.com/jupyter/accessibility">jupyter/accessibility</a>) </li>
    <li>User feedback</li>
</ul>
<h3>Evaluation report</h3>
<p>
	An evaluation for
	<span class="basic-information website-name">JupyterLab</span>
	is available at:
	<a class="technical-information related-evidence" href="https://github.com/jupyterlab/jupyterlab/issues/9399">https://github.com/jupyterlab/jupyterlab/issues/9399</a>.
</p>
<h3>Evaluation report</h3>
<p>
	User reports on JupyterLab's accessibility are available at:
	<a class="technical-information related-evidence" href="https://github.com/jupyterlab/jupyterlab/labels/tag%3AAccessibility">https://github.com/jupyterlab/jupyterlab/labels/tag%3AAccessibility</a>.
</p>
<h3>Formal complaints</h3>
<p class="complaints">For formal complaints, please <ul>
    <li><a href="https://github.com/jupyter/accessibility/issues/new">Write an issue on jupyter/accessibility</a>.</li><li>Write an issue on jupyterlab/jupyterlab and request it be labeled <a href="https://github.com/jupyterlab/jupyterlab/labels/tag%3AAccessibility">tag:accessibility.</a></li></ul></p>
<hr noshade="noshade">
<h3>Date</h3>
<p>
	This statement was created on
	<span class="basic-information statement-created-date">16 May 2022</span>
	using the <a href="https://www.w3.org/WAI/planning/statements/">W3C Accessibility Statement Generator Tool</a>.
</p>

## Things I think we should add

From [Make your website or app accessible and publish an accessibility statement  - Gov.UK](https://www.gov.uk/guidance/make-your-website-or-app-accessible-and-publish-an-accessibility-statement), there are a few things we need to do. I've added check boxes to the quote to show what I think is covered in the above draft.
> Your statement needs to cover:
> - [x] whether your website or mobile app is ‘fully’, ‘partially’ or ‘not’ compliant with accessibility standards
> - [x] if it’s not fully compliant, which parts of your website or mobile app do not currently meet accessibility standards and why (for example, because they are exempt or it would be a disproportionate burden to fix things)
> - [ ] how people can get alternatives to content that’s not accessible to them
> - [x] how to contact you to report accessibility problems - and a link to the website that they can use if they’re not happy with your response.

With this and the other resources, I think we may want to add
- How people can get alternatives to content that's not accessible to them (mentioned above).
- An [accessibility monitoring methodology](https://www.gov.uk/guidance/accessibility-monitoring-how-we-test) section (ie. the testing work). This could be a link to a document that goes more into detail about this process since I doubt it is the main focus for everyone reviewing this statement.
- Maybe a note about open source. We'd have to be sure not to let this sound like an excuse, but rather a note about response times and that there's no support for this work.
- A way for people to report they'd like to be reached out if/when we are testing (like [Gatsby's Surveys and Research section](https://www.gatsbyjs.com/accessibility-statement/#surveys-and-research-projects)). Since I think this could come off wrong, we may want to mention like "for outreach in any potential paid testing opportunities." At the same time, it'd be useful to know we have people available if we want to apply for funding this kind of work in the future.

## What next

To move forward, I would like to know a few things.
- What seems to be inaccurate or missing from this current draft?
- What's on your wish list of statement additions?
- How would we like to collaborate on this next? Is it a good time to open a PR, or do y'all find it easier to iterate elsewhere first?
- I filled this out for JupyterLab. Is there any supporting project we think should be included (ex. the notebook file)?

**Comment by trallard** [comment]
Other resources I have mentioned before are the [gov.uk statement](https://www.gov.uk/help/accessibility-statement) and their [model a11y statement](https://www.gov.uk/guidance/model-accessibility-statement)

The first document has a `PDFs and non-HTML documents` section - do we have non HTML content we need to add? Does this cover the question

>How people can get alternatives to content that's not accessible to them (mentioned above).

> An [accessibility monitoring methodology](https://www.gov.uk/guidance/accessibility-monitoring-how-we-test) section (ie. the testing work). This could be a link to a document that goes more into detail about this process since I doubt it is the main focus for everyone reviewing this statement.

yes - we need to spend some time restructuring the a11y site - to add new sections one of this being the `how we test` and the `statement` plus this ties with the work you have been doing in #95 since I do want to add those scripts there

> Maybe a note about open source. We'd have to be sure not to let this sound like an excuse, but rather a note about response times and that there's no support for this work.
Yes... and an invitation to folks to contribute which ties with the last point about reaching out 

TODOs
- open the PR
- all review
- create an issue template in jupyter/accessibility for the complaints (this might tie with broader improvements the repo needs but this for now)

**Comment by isabela-pf** [comment]
The first draft PR has been made in [quansight-labs/accessibility #3](https://github.com/Quansight-Labs/accessibility/pull/3). Once it is approved there it will become a PR to jupyter/accessibility.

**Comment by isabela-pf** [comment]
PR is up at [jupyter/accessibility #94](https://github.com/jupyter/accessibility/pull/94).

**Comment by isabela-pf** [comment]
I think this should be closed either
- Now, considering this has been created and contributed with some feedback
- Once it is in the main JupyterLab repo (or wherever it needs to live to be linked with that software)

I'm open to either.
